### PR TITLE
Create product suggest endpoint

### PR DESCRIPTION
**Pull request by j-rods**
# Description

Create product suggest GET endpoint
  
# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made my code self-documenting where possible, with the use of clear variable and function names
- [ ] I have commented the reasons for any complex or unusual code, so others can understand the context
- [ ] I have made corresponding changes to the documentation
- [ ] I have added relevant tests for any new functionality I have added
- [x] I have updated the CHANGELOG
- [ ] My changes do not generate any new warnings, either in tooling or in the browser console

### CODE REVIEW

File: src/Resources/Products/suggest.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import Client from '../../Client';
+import AbstractResource from '../AbstractResource';
+import Associations from './Associations';
+import { HttpClientConfig, HttpClientResponse } from '../../Interfaces/HttpClient';
+
+export default class Products extends AbstractResource {
```
Comment (BellaT) [inline src/Resources/Products/suggest.ts:6]: This needs to reflect the file name

File: src/Resources/Products/suggest.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public associations: Associations;
```
Comment (BellaT) [inline src/Resources/Products/suggest.ts:7]: Why is Associations being used here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -59,6 +61,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add orders endpoints ([#212](https://github.com/whirli/whirli-client/pull/212))
 - Add checkout endpoint ([#213](https://github.com/whirli/whirli-client/pull/213))
 - Add shipping addon resource and get all endpoint ([#214](https://github.com/whirli/whirli-client/pull/214))
+### 🐛 Bug fixes
+- Change returnOrder complete from PUT to POST to test image uploading ([#195](https://github.com/whirli/whirli-client/pull/195))
```
Comment (BellaT) [inline CHANGELOG.md:65]: Why is this added here?

File: src/Resources/Products/Suggest.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+export default class Suggest extends AbstractResource {
+    initialise(): void {
+        this.resourcePath = '';
+        this.defaultAccess = 'guest';
+    }
+
+    public all: Function = this.createMethodFromPartialSpec({
+        method: 'GET',
+        path: '/products/suggest',
```
Comment (BellaT) [inline src/Resources/Products/Suggest.ts:11]: I think better to have this in the `this.resourcePath` above for in case more methods are added later on

File: CHANGELOG.md (hunk 3 above)
Comment (j-rods) [reply CHANGELOG.md:65]: I had a crazy merge conflict, let me have a look at whats in develop

Comment (j-rods) [reply CHANGELOG.md]: this is fixed now
